frontend/test/pytest/test_gradient.py

```diff
@@ -246,6 +246,46 @@ def f(x: float):
     assert np.allclose(result, expected)
 
 
+def test_value_and_grad_on_qjit_classical_vector():
+    """Check that value_and_grad works when called on an qjit object that does not wrap a QNode
+    and takes in a vector.
+    """
+
+    @qjit
+    def f(vec):
+        # Takes in a 2D vector (x,y) and computes 30x+40y
+        prod = jnp.array([30, 40]) * vec
+        return prod[0] + prod[1]
+
+    x = jnp.array([1.0, 1.0])
+    result = qjit(value_and_grad(f))(x)
```

Comment:
@paul0403 since you are applying `qjit` here, do you need the `@qjit` decorator above?

Reply:
I think this is the same usage convention as some of the other tests in the file; and also we'd like to make sure the `value_and_grad` itself is inside a qjit context, so it does not get dispatched to jax.value_and_grad.